File: hw/maisincho/u3/HW18Singleton/Singleton/src/main/java/ec/edu/espe/singleton/view/Singleton.java
package ec.edu.espe.singleton.view;

import ec.edu.espe.singleton.controller.Calculator;
import ec.edu.espe.singleton.model.USTax;

/**
 *
 * @author Isaac Maisincho Crafters_Market DCCO ESPE
 */
public class Singleton {

    public static void main(String[] args) {
        Calculator calculator = new Calculator(); 

        float subtotal = 10.0F;
        float total = calculator.salesTotal(subtotal);
        System.out.println("The total is --> $" + total);

        USTax.getInstance().setTaxPercentage(0.15F);
        total = calculator.salesTotal(subtotal);
        System.out.println("The total with 15% tax is --> $" + total);
    }
}

File: hw/maisincho/u3/HW18Singleton/SingletonPy/Singleton.py
from controller.Calculator import Calculator
from model.USTax import USTax

def main():
    calculator = Calculator()

    subtotal = 10.0
    total = calculator.sales_total(subtotal)
    print(f"The total is --> ${total}")

    USTax.get_instance().set_tax_percentage(0.15)
    total = calculator.sales_total(subtotal)
    print(f"The total with 15% tax is --> ${total}")

if __name__ == "__main__":
    main()